File: content/problems/0088-merge-sorted-array/solutions/reverse-two-pointers/Solution.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int first = m - 1, second = n - 1, write = m + n - 1;
        while (second >= 0) nums1[write--] = first >= 0 && nums1[first] > nums2[second] ? nums1[first--] : nums2[second--];
    }
}

File: content/problems/0088-merge-sorted-array/solutions/reverse-two-pointers/Solution.kt
class Solution {
    fun merge(nums1: IntArray, m: Int, nums2: IntArray, n: Int) {
        var first = m - 1; var second = n - 1; var write = m + n - 1
        while (second >= 0) nums1[write--] = if (first >= 0 && nums1[first] > nums2[second]) nums1[first--] else nums2[second--]
    }
}
